PukingPredator/Assets/Scripts/Eating.cs

```diff
@@ -65,26 +71,46 @@ private void GameInput_Eat()
         {
             // Get the GameObject that was hit
             GameObject hitObject = hit.collider.gameObject;
-
-            var consumableData = hitObject.GetComponent<Consumable>();
-            if (consumableData == null || !consumableData.isConsumable) { return; }
-
-            inventory.PushItem(consumableData);
-            consumableData.SetState(ItemState.beingConsumed);
+            Process_Consumption(hitObject);
         }
-    }
+        else if (Physics.SphereCast(ray, radius: 0.2f, out hit, maxDistance: 2f, layerMask: dashLayers))
+        {
+            // Dashes if the object is far
+            GameObject hitObject = hit.collider.gameObject;
+            Process_Consumption(hitObject);
+
+            Player player = player = this.GetComponent<Player>();
+            if (player != null)
+            {
+                player.SetTarget(hitObject.transform.position);
+                player.SetState(MovementState.eating);
+            }
+        }
+    }
 
     private void GameInput_Puke()
     {
         if (inventory.isEmpty) { return; }
 
         Consumable itemToPlace = inventory.PopItem();
-
+        if (itemToPlace == null) { return; } // case when you press puke and eat at the same time
         var pukeDir = transform.forward;
         var targetPosition = transform.position + pukeDir * pukeDistance;
         itemToPlace.PlaceAt(targetPosition);
     }
 
+    /// <summary>
+    /// Consumes the object and updates the inventory
+    /// </summary>
+    private void Process_Consumption(GameObject hitObject)
```

Comment:
id probably swap this to:
private void ConsumeObject(GameObject obj)
... (or ConsumeItem). The current name is inconsistent with other function names, the only ones that use an underscore thus far are "GameInput_[the input]" (which arguably also shouldn't use one, but that shouldn't be changed here). and also "hitObject" doesnt fit as an input name because the code inside the function doesnt care about how the item is selected anymore (a collision/hit), its just about consuming it